src/python/pants/backend/go/util_rules/third_party_pkg.py

```diff
@@ -111,14 +106,13 @@ def debug_hint(self) -> str:
 
 @dataclass(frozen=True)
 class AllThirdPartyPackages(FrozenDict[str, ThirdPartyPkgAnalysis]):
-    """All the packages downloaded from a go.mod, along with a digest of the downloaded files.
+    """All the packages downloaded from a go.mod.
 
-    The digest has files in the format `gopath/pkg/mod`, which is what `GoSdkProcess` sets `GOPATH`
-    to. This means that you can include the digest in a process and Go will properly consume it as
-    the `GOPATH`.
+    There is no longer a digest included, as the packages are downloaded as to a named_cache. Later
+    processes may use the file contents via append_only_caches=sdk.sdk_cache(). The simlink will
+    create the gopath such that go will properly consume it.
     """
 
-    digest: Digest
```

Comment:
Would removal of `digest` here have an impact on my WIP `go mod vendor` PR? https://github.com/pantsbuild/pants/pull/18175. Probably not since that PR ends up setting `digest` to `EMPTY_DIGEST` so even in that PR this field is moot. https://github.com/pantsbuild/pants/pull/18175/files#diff-0dc70bceca45921afbd472d87a5d4652e5ee5b446fc21bc9c352abc49a99fc28R782

Reply:
Agreed that I don't think it breaks that PR.  But the main benefit of dropping the field was allowing type checks to help me find and ensure that downstream code would function even with an EMPTY_DIGEST.  So, let me know if you'd like the digest added back.